packages/@aws-cdk/aws-appsync/test/appsync.test.ts

```diff
@@ -191,3 +192,37 @@ test('appsync GraphqlApi should be configured with custom domain when specified'
     DomainName: domainName,
   });
 });
+
+test('log retention should be configured with given retention time when specified', () => {
```

Comment:
Could we get a testcase that tests for it expected output when it's not provided? Even if that just means it's missing from the template.

Reply:
Yes, you can, that was my original idea but I couldn't bend my head around the idea of asserting something that doesn't exist 😓 The easiest option that comes to my mind would be inverting `hasResource` but I think that is not possible?

Reply:
How about `Match.absent()` ? https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.assertions-readme.html#presence-and-absence

```
    logConfig: {
      retention: Match.absent(),
    },
```

Reply:
Hmm... `logConfig.retention` is not a property of the `AWS::AppSync::GraphQLApi`. For sure I can add a test that checks it but does it make sense?

Reply:
It would be more interesting to check that `Custom::LogRetention` won't be created when `logConfig.retention` is not defined but how can I do that of `hasResource` cannot be be inverted?

Reply:
@TheRealAmazonKendra, I think this is the only open question on this pull request, and I think it is your call to choose what I suppose to do for it, given the condition I mention above

Reply:
Would `template.resourceCountIs('Custom::LogRetention', 0);` work here?

Reply:
Maybe? Let me try :)

Reply:
It does, thank you @mrgrain! 🙇🏼‍♂️

Reply:
Regardless, I would be happy to see a jest-like `.not` operator in CDK assertions 🤓

Reply:
Agreed! I've been think about that as well. 😃  You know that PRs are always welcome. 😼